api/src/main/java/io/opencensus/stats/BucketBoundaries.java

```diff
@@ -46,14 +50,37 @@ public static final BucketBoundaries create(List<Double> bucketBoundaries) {
     List<Double> bucketBoundariesCopy = new ArrayList<Double>(bucketBoundaries); // Deep copy.
     // Check if sorted.
     if (bucketBoundariesCopy.size() > 1) {
-      double lower = bucketBoundariesCopy.get(0);
+      double previous = bucketBoundariesCopy.get(0);
       for (int i = 1; i < bucketBoundariesCopy.size(); i++) {
         double next = bucketBoundariesCopy.get(i);
-        Utils.checkArgument(lower < next, "Bucket boundaries not sorted.");
-        lower = next;
+        Utils.checkArgument(previous < next, "Bucket boundaries not sorted.");
+        previous = next;
       }
     }
-    return new AutoValue_BucketBoundaries(Collections.unmodifiableList(bucketBoundariesCopy));
+    return new AutoValue_BucketBoundaries(
+        Collections.unmodifiableList(dropNegativeBucketBounds(bucketBoundariesCopy)));
+  }
+
+  private static List<Double> dropNegativeBucketBounds(List<Double> bucketBoundaries) {
```

Comment:
Metrics histogram does not accept 0 https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/metrics/export/Distribution.java#L211 so I think you need to drop the 0 as well but don't record an error for that (or maybe just a LOG.info if you really care) because that is correctly handled by metrics with the default [0, first_item) interval.